src/Microsoft.AspNet.Mvc.Core/ModelBinders/BodyModelBinder.cs

```diff
@@ -51,8 +51,13 @@ public BodyModelBinder()
             {
                 var model = await formatter.ReadAsync(formatterContext);
 
-                // key is empty to ensure that the model name is not used as a prefix for validation.
-                return new ModelBindingResult(model, key: string.Empty, isModelSet: true);
+                var isTopLevelObject = bindingContext.ModelMetadata.ContainerType == null;
```

Comment:
Didn't we conclude in another discussion that this kind of logic is incorrect, and we can only really know this by adding it back to the context?